packages/mobile/src/pincode/authentication.test.ts

```diff
@@ -80,3 +81,52 @@ describe(getPincode, () => {
     expect(getCachedPin(DEFAULT_CACHE_ACCOUNT)).toBeNull()
   })
 })
+
+describe(PinBlocklist, () => {
+  const blocklist = new PinBlocklist()
+
+  describe('#contains', () => {
+    const commonPins = [
+      '000000',
+      '123456',
+      '111111',
+      '123123',
+      '159951',
+      '007007',
+      '110989',
+      '789789',
+      '456456',
+      '852456',
+      '999999',
+    ]
+
+    for (const commonPin of commonPins) {
+      it(`indicates the list contains common PIN ${commonPin}`, () => {
+        expect(blocklist.contains(commonPin)).toBe(true)
+      })
+    }
+
+    it('indicates inclusion of a small portion of random PINs', () => {
+      // Using the frequentist estimator of true probability for a Bernoulli process.
+      // https://en.wikipedia.org/wiki/Checking_whether_a_coin_is_fair#Estimator_of_true_probability
+      // Using 2000 trials, and a confidence interval Z value of 3.89 gives the test a 1 in 10,000
```

Comment:
If I understand this correctly, will this test fail once every 10k runs (on average)?

If so, can we just run it again if it fails? The probability of it failing twice in a row should be significantly lower, right? 1 in 10,000 is not high but also not completely insignificant either.

Reply:
Indeed, if the test is rerun it will have would only have a 1 in 10,000 chance of failing a second time. (So 1 in 100,000,000 of failing both times)